lib/dialects/postgres/data-types.js

```diff
@@ -7,6 +7,20 @@ const inherits = require('../../utils/inherits');
 module.exports = BaseTypes => {
   const warn = BaseTypes.ABSTRACT.warn.bind(undefined, 'http://www.postgresql.org/docs/9.4/static/datatype.html');
 
+  /**
+   * Removes unsupported Postgres options, i.e., LENGTH, UNSIGNED and ZEROFILL, for the integer data types.
+   * @param dataType The base integer data type.
+   */
+  function removeUnsupportedOptions(dataType) {
+    if (this._length || this.options.length || this._unsigned || this._zerofill) {
+      warn('SQLite does not support ' + dataType._name + '  with LENGTH, UNSIGNED or ZEROFILL. Plain `' + dataType._name + '` will be used instead.');
```

Comment:
`PostgreSQL does not support <typename> with options. Plain '<typename>' will be used instead.`

Reply:
Yes I see. What do you think if I define an util file like this:

```
'use strict';

/**
 * Removes unsupported options, i.e., LENGTH, UNSIGNED and ZEROFILL, for the integer data types.
 *
 * @param dialect The specific SQL dialect
 * @param dataType The base integer data type.
 * @param warn The dialect specific warn function.
 */
function removeUnsupportedOptions(dialect, dataType, warn) {
  let warnMessage = '';

  if (this._length || this.options.length || this._unsigned || this._zerofill) {
    this._unsigned = undefined;
    this._zerofill = undefined;

    if (dialect === 'sqlite') {
      warnMessage = 'SQLite does not support ' + dataType._name + '  with UNSIGNED or ZEROFILL. ';
    } else if (dialect === 'postgres') {
      warnMessage = 'PostgresSQL does not support ' + dataType._name + '  with LENGTH, UNSIGNED or ZEROFILL. ';
      this._length = undefined;
      this.options.length = undefined;
    }
    
    warn(warnMessage + 'Plain `' + dataType._name + '` will be used instead.');
  }
}


module.exports = removeUnsupportedOptions;
```

Reply:
No that will overtime become complex to handle different dialect and their particularities. Keep these functions in `data-types.js` of those dialects, so they can handle dialect specific changes

Also rename `removeUnsupportedOptions` => `removeUnsupportedIntegerOptions`

Reply:
Sorry, dataType instead of this.

Reply:
Done.